File: terre08.js
// Racine carré d'un nombre
//
if (process.argv.length <= 2) {
    console.log("Usage: terre08.js <number>")
    process.exit(1)
}

const arg = process.argv[2]
const number = parseInt(arg)

if (isNaN(number)) {
    console.log(`Expected a valid number got=${arg}`)
    process.exit(1)
}

if (number < 0) {
    console.log(`Expected a positive number got=${arg}`)
    process.exit(1)
}

let i = 1
while (i * i <= number) {
    i++
}

console.log(--i)

File: terre08.exs
case System.argv() do
  [arg] ->
    case Integer.parse(arg) do
      :error ->
        IO.puts("Expected a valid number got=#{arg}")
        System.stop(1)

      {number, _rest} when number < 1 ->
        IO.puts("Expected a positive number")
        System.stop(1)

      {number, _rest} ->
        1..number
        |> Enum.find(fn divisor ->
          divisor * divisor > number
        end)
        |> then(fn divisor -> divisor - 1 end)
        |> IO.puts()
    end

  _args ->
    IO.puts("Usage: elixir terre08.exs")
    System.stop(1)
end
